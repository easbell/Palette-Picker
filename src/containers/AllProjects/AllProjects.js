import React from 'react'
import DisplayProject from '../../components/DisplayProject/DisplayProject'
import { connect } from 'react-redux'
import './AllProjects.css'

export const AllProjects = (props) => {
    
    const projects = props.projects.map(project => {
      return <DisplayProject key={project.id} project={project}/>
    })
    
    return(
      <div className='projects-container'>
        {projects}
      </div>
    )
}

const mapStateToProps = (state) => ({
  projects: state.projects
})

export default connect(mapStateToProps)(AllProjects)