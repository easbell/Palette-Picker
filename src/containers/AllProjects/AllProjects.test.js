import React from 'react'
import { AllProjects, mapStateToProps } from './AllProjects'
import { shallow } from 'enzyme'

describe('AllProjects', () => {
  let wrapper
  let mockProjects = [{ id: 1, name: 'my project'}, {id: 2, name: 'my other project'}]

  it('should match the snapshot', () => {
    wrapper = shallow(<AllProjects projects={mockProjects} />)
    expect(wrapper).toMatchSnapshot()
  })

  describe('mapStateToProps', () => {
    it('returns correct props', () => {
      const mockState = {
        projects: ['projects']
      }
      const expectedProps = {
        projects: ['projects']
      }

      const mappedProps = mapStateToProps(mockState);
      expect(mappedProps).toEqual(expectedProps)
    });
  });
})