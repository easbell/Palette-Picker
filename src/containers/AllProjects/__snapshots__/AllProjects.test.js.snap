// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AllProjects should match the snapshot 1`] = `
<div
  className="projects-container"
>
  <ConnectFunction
    key="1"
    project={
      Object {
        "id": 1,
        "name": "my project",
      }
    }
  />
  <ConnectFunction
    key="2"
    project={
      Object {
        "id": 2,
        "name": "my other project",
      }
    }
  />
</div>
`;
